Tasks/VsTestV3/runvstest.ts

```diff
@@ -190,6 +193,25 @@ function isServerBasedRun(): boolean {
     return false;
 }
 
+function setUpConnectedServiceEnvironmentVariables() {
+    var connectedService = tl.getInput('ConnectedServiceName');
+    if(connectedService) {
+        var authScheme: string = tl.getEndpointAuthorizationScheme(connectedService, false);
+        if (authScheme && authScheme.toLowerCase() == "workloadidentityfederation") {
+            process.env.AZURESUBSCRIPTION_SERVICE_CONNECTION_ID = connectedService;
```

Comment:
Maybe you also would like to remove these values from environment once the task completes. Similar to what is done in AzureCLI 2 [here](https://github.com/microsoft/azure-pipelines-tasks/blob/ebb77fe381b29bccd2e0cab033f25ca8df7a62d8/Tasks/AzureCLIV2/azureclitask.ts#L180-L185).

Reply:
I was under the impression that the environment variables are not shared across tasks. I did validate that via a sample yaml.

```
- task: VSTest@3
      inputs:
        azureSubscription: 'TestServiceConnection'
        testSelector: 'testAssemblies'
        testAssemblyVer2: |
          **\bin\**\UnitTestProject1.dll
        searchFolder: '$(System.DefaultWorkingDirectory)'
        runSettingsFile: 'cc.runsettings'
        codeCoverageEnabled: true
        platform: '$(buildPlatform)'
        configuration: '$(buildConfiguration)'
      env:
        SYSTEM_ACCESSTOKEN: $(System.AccessToken)

- task: VSTest@3
      inputs:
        testSelector: 'testAssemblies'
        testAssemblyVer2: |
          **\bin\**\UnitTestProject1.dll
        searchFolder: '$(System.DefaultWorkingDirectory)'
        runSettingsFile: 'cc.runsettings'
        codeCoverageEnabled: true
        platform: '$(buildPlatform)'
        configuration: '$(buildConfiguration)'
      env:
        SYSTEM_ACCESSTOKEN: $(System.AccessToken)
```
Second task failed saying that AZURESUBSCRIPTION_SERVICE_CONNECTION_ID is NULL.